crates/primitives/src/proofs.rs

```diff
@@ -69,13 +69,20 @@ pub fn calculate_withdrawals_root(withdrawals: &[Withdrawal]) -> B256 {
 }
 
 /// Calculates the receipt root for a header.
-pub fn calculate_receipt_root(receipts: &[ReceiptWithBloom]) -> B256 {
+pub fn calculate_receipt_root(
+    receipts: &[ReceiptWithBloom],
+    #[cfg(feature = "optimism")] chain_spec: std::sync::Arc<crate::ChainSpec>,
+    #[cfg(feature = "optimism")] timestamp: u64,
+) -> B256 {
```

Comment:
I'd like this as a separate function for OP instead

this will make it easier once we eventually remove the feature